### PR TITLE
Plugin release docs: Multiple changes

### LINKED ISSUE #50343: Update plugin release documentation
Updates to the [plugin release documentation:](https://developer.wordpress.org/block-editor/contributors/code/release/)

- [x] Remove reference to Performance tests
- [x] Add more details to the 'npm run other:cherry-pick "Backport to Gutenberg RC"' section to favor it over the series of git commands. 

### Screenshots, screen recording, code snippet

Screenshot 2023-05-03 at 11 26 30

### PR DESCRIPTION
**Pull request by bph**
- Adds cherry picking via npm command + screenshot
- adds more changelog curation details and moves it to the RC section. 

Fixes: #50343 #49884 #42516
## Why?
Streamline the process, better onboard new contributors and update on the process.

### CODE REVIEW

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -107,23 +164,7 @@ Documenting the release is a group effort between the release manager, Gutenberg
 
 #### 1. Curating the changelog
```
Comment (priethor) [inline docs/contributors/code/release.md]: This section is empty except for a `T`

File: docs/contributors/code/release.md [hunk 2]
```diff
@@ -47,16 +47,51 @@ This message is misleading and no action is required by the release coordinator.
 
 As soon as the workflow has finished, you'll find the release draft under [Gutenberg Releases](https://github.com/WordPress/gutenberg/releases). The draft is pre-populated with changelog entries based on previous release candidates for this version and any changes that have since been cherry-picked to the release branch. Thus, when releasing the first stable version of a series, delete any RC version headers (that are only there for your information) and move the more recent changes to the correct section (see below).
 
-The changelog draft will be partially pre-organized (based on GitHub labels) into sections and, within those, into “features.” Take some time to read the generated notes and then edit them to ensure legibility and accuracy.
+## Streamline curating the release changelog.
```
Comment (priethor) [inline docs/contributors/code/release.md:83]: I think this section could be split into more digestible, smaller ones.

**Review by priethor — COMMENTED** [review]
Thanks, Birgit. Rather than expanding the documentation, we should aim at simplifying it. Currently, it kind of feels overwhelming for first-timers, as it feels like one should read and understand the whole documentation to wrangle a release. In contrast, some of it (e.g., cherry-picking) only needs to be consulted on a need-to-know basis.

For starters, apart from shortening information about the launch process itself (it's just a GitHub action), we should further divide the document into sections and add a Table of Content on top. This can be done in follow-up PRs, though.

File: docs/contributors/code/release.md (hunk 2 above)
Comment (bph) [reply docs/contributors/code/release.md:83]: I have: 

- Review of Milestone Labels
- Curating the changelog markdown. 
I will edit it for brevity soon.

File: docs/contributors/code/release.md (hunk 1 above)
Comment (bph) [reply docs/contributors/code/release.md]: Fixed

**Comment by bph** [comment]
@priethor Thanks for reviewing the draft PR. 

There are things that don't take any effort at all, like the workflow. They still need to be explained and are covered. Some things take a lot of time if want to make the most of the time spent and have a good outcome, like curating the changelog and creating the release post. 

Now, it would probably be helpful if we create the release documentation [following the issue checklist](https://github.com/WordPress/gutenberg/issues/50970). So each item that needs _some_ explanation, is covered, in the order it's worked on.

**Comment by priethor** [comment]
> Now, it would probably be helpful if we create the release documentation https://github.com/WordPress/gutenberg/issues/50970. So each item that needs some explanation, is covered, in the order it's worked on.

I like that idea!

**Comment by bph** [comment]
@ndiego and I are continuing [our work on Google Docs](https://docs.google.com/document/d/17eiwqv-hSTGphI85BustyuGWxn6jdm4mrUyeMAlX0mo/edit?usp=sharing) for now.

**Comment by bph** [comment]
Closed as replaced by #52955

**Pull request closed without merging**